temporalio/workflow.py

```diff
@@ -1230,22 +1247,29 @@ def process(
         if (
             self.workflow_info_on_message
             or self.workflow_info_on_extra
+            or self.update_info_on_extra
             or self.full_workflow_info_on_extra
         ):
+            extra: Dict[str, Any] = {}
+            msg_extra: Dict[str, Any] = {}
             runtime = _Runtime.maybe_current()
             if runtime:
                 if self.workflow_info_on_message:
-                    msg = f"{msg} ({runtime.logger_details})"
+                    msg_extra.update(runtime.logger_details)
                 if self.workflow_info_on_extra:
-                    # Extra can be absent or None, this handles both
-                    extra = kwargs.get("extra", None) or {}
                     extra["temporal_workflow"] = runtime.logger_details
-                    kwargs["extra"] = extra
                 if self.full_workflow_info_on_extra:
-                    # Extra can be absent or None, this handles both
-                    extra = kwargs.get("extra", None) or {}
                     extra["workflow_info"] = runtime.workflow_info()
-                    kwargs["extra"] = extra
+            update_info = current_update_info()
+            if update_info:
+                if self.update_info_on_extra:
+                    extra["temporal_update"] = asdict(update_info)
```

Comment:
Arguably update info is workflow info and doesn't need a separate top-level key. I think it can just have the extra fields be put into `temporal_workflow` section same as appending to the message does.